fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java

```diff
@@ -167,6 +168,9 @@ public boolean equals(Object o) {
             return false;
         }
         GroupExpression that = (GroupExpression) o;
+        if (plan instanceof LogicalOlapScan) {
```

Comment:
You should add a comment here to show why LogicalOlapScan is not equals

Reply:
Ok，I add it now.